source/agora/network/Manager.d

```diff
@@ -343,7 +343,7 @@ public class NetworkManager
             while (1)
             {
                 scope (success)
-                    this.outer.taskman.wait(this.outer.node_config.retry_delay);
+                    this.outer.taskman.wait(this.outer.node_config.network_discovery_interval);
```

Comment:
I feel like the right aproach would be something like;

```
while (1)
{
   foreach (client; clients)
   {
      // work
     wait(retry_delay); // totally optional
   }
   wait(network_discovery_interval);
}
```

Otherwise it would a really long time to go thru all the clients

Reply:
Updated